templates/activity-stream/impression-stats/impression-stats.1.schema.json

```diff
@@ -0,0 +1,83 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "title": "impression-stats",
+  "properties": {
+    "impression_id": {
+      "type": "string",
+      "pattern": "^\\{[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\\}$"
+    },
+    "shield_id": {
+      "type": "string"
+    },
+    "sample_id": {
```

Comment:
Correction: `sample_id` is *not* included in the payload from client. Instead, it's a random integer auto generated by the database. My apologies for the confusion. How can we handle this special case? I.e. Having a default (or auto generated) field in the DB schema.

Reply:
> How can we handle this special case? I.e. Having a default (or auto generated) field in the DB schema, but not in the payload schema.

A static default value could be handled in a SQL view definition, same as renaming a field. A SQL view could also handle running a hash function to stably generate a sample_id from properties of the message itself, as we do for telemetry where we hash client_id to get sample_id.

The case of the database auto-generating a random value, though, would require a periodic scheduled ETL job that inserts records into a destination table.